### PR TITLE
Fix markSimplified usage

### PR DESCRIPTION
**Pull request by virgil-serbanuta**
---

###### Reviewer checklist

- [ ] Test coverage: `stack test --coverage`
- [ ] Public API documentation: `stack haddock`

### CODE REVIEW

File: kore/src/Kore/Step/Simplification/Exists.hs [hunk 1]
```diff
@@ -253,8 +282,10 @@ makeEvaluateBoundRight variable freeSubstitution normalized = do
     TopBottom.guardAgainstBottom simplifiedPattern
     return simplifiedPattern
   where
+    simplifiedPattern :: Pattern variable
     simplifiedPattern =
-        Conditional.andCondition
+        Pattern.markSimplified
```
Comment (ttuegel) [inline kore/src/Kore/Step/Simplification/Exists.hs]: Does this eliminate the need to `markSimplified` in `quantifyPattern` below? How do we know all parts of the pattern are really simplified?

Comment (virgil-serbanuta) [reply kore/src/Kore/Step/Simplification/Exists.hs]: You're right, we don't, I'll fix that.

File: kore/src/Kore/Internal/Pattern.hs [hunk 2]
```diff
@@ -84,6 +95,11 @@ isSimplified :: Pattern variable -> Bool
 isSimplified (splitTerm -> (t, p)) =
     TermLike.isSimplified t && Predicate.isSimplified p
 
+markSimplified :: Pattern variable -> Pattern variable
```
[the remark this answers is not in the document]
Comment (virgil-serbanuta) [reply kore/src/Kore/Internal/Pattern.hs]: Removed

File: kore/src/Kore/Step/Simplification/Exists.hs (hunk 1 above)
Comment (virgil-serbanuta) [reply kore/src/Kore/Step/Simplification/Exists.hs]: Fixed.

**Pull request merged**